### PR TITLE
Remove redundant DiagnosticAnalyzer attributes

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**

### CODE REVIEW

**Review by csaba-sagi-sonarsource — CHANGES_REQUESTED** [review]
Do you mind adding a couple of UTs to https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/tests/SonarAnalyzer.UnitTest/Common/RuleTest.cs that make sure that classes derived from SymbolicRuleCheck do not have this attribute and one that checks that the classes that have them derive from the correct base class?

**Comment by costin-zaharia-sonarsource** [comment]
I don't mind at all :) I even think we should not restrict only to `SymbolicRuleCheck derived` classes. We can make the check more general to check that all the classes that have the attribute derive from DiagnosticAnalyzer.

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM!

File: analyzers/tests/SonarAnalyzer.UnitTest/Common/RuleTest.cs [hunk 1]
```diff
@@ -72,6 +72,15 @@ public void AllAnalyzers_InheritSonarDiagnosticAnalyzer()
             }
         }
 
+        [TestMethod]
+        public void AllClassesWithDiagnosticAnalyzerAttribute_InheritSonarDiagnosticAnalyzer()
```
Comment (pavel-mikula-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Common/RuleTest.cs:76]: Nice

**Pull request merged**